tests/unit/conditions/test_json_condition.py

```diff
@@ -0,0 +1,194 @@
+import json
+
+import pytest
+
+from nucypher.policy.conditions.exceptions import InvalidCondition
+from nucypher.policy.conditions.json.json import JsonCondition
+from nucypher.policy.conditions.lingo import ConditionLingo, ReturnValueTest
+
+
+def test_json_condition_initialization():
+    data = {"store": {"book": [{"price": 10.5}]}}
+    condition = JsonCondition(
+        data=data,
+        query="$.store.book[0].price",
+        return_value_test=ReturnValueTest("==", 10.5),
+    )
+    assert condition.data == data
+    assert condition.query == "$.store.book[0].price"
+
+
+def test_json_condition_with_json_string():
+    json_string = '{"value": 42}'
+    condition = JsonCondition(
+        data=json_string,
+        query="$.value",
+        return_value_test=ReturnValueTest("==", 42),
+    )
+    assert condition.data == {"value": 42}
+
+
+def test_json_condition_with_primitive():
+    condition = JsonCondition(
+        data=42,
+        return_value_test=ReturnValueTest("==", 42),
+    )
+    assert condition.data == 42
+    assert condition.query is None
```

Comment:
Related to my comment above. I know that `"42"` is a valid JSON string, but `42` is just an integer.

Reply:
I thnk `42` is also valid json, it's certainly accepted by a lot of json parsers, included the one we use

Reply:
Right, my point was more about acceptable types (str vs integers), but since you mentioned before that with sequential conditions JSON data may be already parsed as objects, then I have no further comment 👍